src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -186,16 +191,15 @@ private static void initProcessingRules() {
                     final List<String> passedThroughInputAtts = Collections.unmodifiableList(Arrays.asList(
                             "rows",
                             "query"
-                    ));
-
-                    p.parseControl((IFormElement) parent, e, Constants.CONTROL_INPUT, passedThroughInputAtts,
-                            passedThroughInputAtts);
+                    ));p.parseControl((IFormElement) parent, e, Constants.CONTROL_INPUT,passedThroughInputAtts,
```

Comment:
Looks like some funny things happened with spacing in the merge!

Reply:
I've fixed that, but that class could benefit of static imports a lot. That block could be:
```java
final List<String> passedThroughInputAtts = unmodifiableList(asList("rows","query"));
p.parseControl((IFormElement) parent, e, CONTROL_INPUT, passedThroughInputAtts, passedThroughInputAtts);
```
(less cluttered)

Shall I make that change too?

Also, why is that list `final`? It's used there only to pass it to the `p.parseControl()` method, so, so what if someone reassigns it?

Reply:
I like static imports where they improve readability and shorten code.

Scala has val and var. We use val unless the identifier’s value is allowed to change (which is generally discouraged). val is like Java’s final. The trouble with final in java is that it’s a keyword, and adds noise. So I generally skip it if it’s immediately obvious to the reader that the value doesn’t change, e.g., when its scope is just a few lines. So, I wouldn’t object to removing this final.